tests/test_model_saving.py

```diff
@@ -440,6 +440,84 @@ def test_loading_weights_by_name_skip_mismatch():
         assert_raises(AssertionError, assert_allclose, old, new, atol=1e-05)
 
 
+def test_weight_saving_to_pre_created_h5py_file():
+    inputs = Input(shape=(3,))
+    x = Dense(2)(inputs)
+    outputs = Dense(3)(x)
+
+    model = Model(inputs, outputs)
+    model.compile(loss=losses.MSE,
+                  optimizer=optimizers.Adam(),
+                  metrics=[metrics.categorical_accuracy])
+    x = np.random.random((1, 3))
+    y = np.random.random((1, 3))
+    model.train_on_batch(x, y)
+
+    out = model.predict(x)
+    _, fname = tempfile.mkstemp('.h5')
+    with h5py.File(fname, mode='r+') as h5file:
+        model.save_weights(h5file)
```

Comment:
This is not a very complete/safe way of testing; it is possible that either `load_weigths` or both `save/load_weights` does nothing and the test would still pass. I'd suggest that you perturb the weights in between the calls for all tests. You can make a local util:
```
def perturb_weights(model):
    model.set_weights([w + 1. for w in model.get_weights()])
``` 
(goas for all tests in this PR)

Reply:
(this is not needed for `load/save_model` because then we compare the output from an entirely new object - here we check the state of the same object)

Reply:
I now always create a fresh model which is loading the weights. thanks for spotting this

Reply:
I now always create a fresh model which is loading the weights. thanks for spotting this